db_service/vcap.go

```diff
@@ -0,0 +1,98 @@
+package db_service
+
+import (
+	"code.cloudfoundry.org/lager"
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/viper"
+	"net/url"
+	"os"
+)
+
+type VcapService struct {
+	BindingName  string            `json:"binding_name"`  // The name assigned to the service binding by the user.
+	InstanceName string            `json:"instance_name"` // The name assigned to the service instance by the user.
+	Name         string            `json:"name"`          // The binding_name if it exists; otherwise the instance_name.
+	Label        string            `json:"label"`         // The name of the service offering.
+	Tags         []string          `json:"tags"`          // An array of strings an app can use to identify a service instance.
+	Plan         string            `json:"plan"`          // The service plan selected when the service instance was created.
+	Credentials  map[string]string `json:"credentials"`   // The service-specific credentials needed to access the service instance.
+}
+
+func useVcapServices(logger lager.Logger) error {
+	vcapData, vcapExists := os.LookupEnv("VCAP_SERVICES")
+	if vcapExists {
```

Comment:
rather than return nil all the way at the bottom if this is false, let's flip the statement around to return early so most of the function isn't in the if-block